### ISSUE TITLE
'在 Linux 操作系统上安装和配置 Flutter 开发环境' page issue

### ISSUE THREAD

**Issue by godot0413301** [cfug/flutter.cn#708]
Page URL: https://flutter.cn/docs/get-started/install/linux.html
Page source: https://github.com/cfug/flutter.cn/tree/master/src/docs/get-started/install/linux.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

Description of issue:
我的操作系统是ubuntu20.04lts桌面版,在中国,使用
`sudo snap install flutter --classic`安装了flutter sdk
已经设置了中国镜像
```
export PUB_HOSTED_URL=https://pub.flutter-io.cn
export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn
```
但是在flutter upgrade升级的时候,还是从https://storage.googleapis.com下载,速度非常慢
```
godot@godot-System-Product-Name:~/桌面$ flutter upgrade
Initializing Flutter
Downloading https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.20.1-stable.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  3  528M    3 18.9M    0     0  92995      0  1:39:15  0:03:33  1:35:42 72082
```

**Comment by godot0413301** [comment]
我的操作系统是ubuntu20.04lts桌面版

**Comment by chenglu** [comment]
+1 @godot0413301 

我在 Ubuntu 20.04 上安装了 snap
```
sudo apt update
sudo apt install snapd
```
并使用
```
sudo snap install flutter --classic
```
log 如下：
```
root@iZ2zebgo0r7mr3igozdclqZ:~# sudo snap install flutter --classic
flutter 0+git.aca15f8 from Flutter Team✓ installed
root@iZ2zebgo0r7mr3igozdclqZ:~# flutter -v
Initializing Flutter
Downloading https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.20.1-stable.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  528M    0 12555    0     0   9964      0 15:26:19  0:00:01 15:26:18  9956
root@iZ2zebgo0r7mr3igozdclqZ:~# export PUB_HOSTED_URL=https://pub.flutter-io.cn
root@iZ2zebgo0r7mr3igozdclqZ:~# export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn
root@iZ2zebgo0r7mr3igozdclqZ:~# flutter -v
Initializing Flutter
Downloading https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.20.1-stable.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  2  528M    2 15.6M    0     0  1623k      0  0:05:33  0:00:09  0:05:24 2318k
```
确实没有从中国镜像下载。

cc/ @InMatrix @xster @liyuqian @zoeyfan:
snap store 里的 Flutter 是不检测环境变量的吗？

**Comment by InMatrix** [comment]
Cc: @csells

TL;DR: the OP reports that the Snap store on Linux seems to ignore the environment variables `PUB_HOSTED_URL` and   `FLUTTER_STORAGE_BASE_URL`. Could you loop in someone from Ubuntu? Thanks!

**Comment by InMatrix** [comment]
@chenglu It might be more helpful to move this issue to https://github.com/kenvandine/flutter-snap/issues

Cc: @kenvandine

**Comment by chenglu** [comment]
OK, I've filed a new issue there (https://github.com/kenvandine/flutter-snap/issues/10), thanks to Tao for the great help!

**Comment by chenglu** [comment]
Hi @godot0413301 

After the lastest commit https://github.com/canonical/flutter-snap/commit/9655f47993953e61652ce3fa08a6074f0a320fac, I think the issue has been resovled, 
I've also tested on an AliCloud ECS, it worked:

```
root@iZ8vbfkm50gffkhlzoe62bZ:~# export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn
root@iZ8vbfkm50gffkhlzoe62bZ:~# export PUB_HOSTED_URL=https://pub.flutter-io.cn
root@iZ8vbfkm50gffkhlzoe62bZ:~# /snap/bin/flutter -v
Initializing Flutter
Downloading https://storage.flutter-io.cn/flutter_infra/releases/stable/linux/flutter_linux_1.20.4-stable.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 24  532M   24  129M    0     0  13.0M      0  0:00:40  0:00:09  0:00:31 11.8M
```

Let us know if you have any further questions!

Thanks to @InMatrix and @MarcusTomlinson!
